app/renderer/components/Inspector/Inspector.js

```diff
@@ -108,6 +129,28 @@ export default class Inspector extends Component {
     this.props.getSavedActionFramework();
     this.props.runKeepAliveLoop();
     window.addEventListener('resize', this.updateSourceTreeWidth);
+
+    if (this.props.mjpegScreenshotUrl) {
+      this.mjpegStreamCheckInterval = setInterval(this.checkMjpegStream.bind(this),
+        MJPEG_STREAM_CHECK_INTERVAL);
+    }
+  }
+
+  async checkMjpegStream () {
+    const {mjpegScreenshotUrl, isAwaitingMjpegStream, setAwaitingMjpegStream} = this.props;
+    const img = new Image();
+    img.src = mjpegScreenshotUrl;
+    let imgReady = false;
+    try {
+      await img.decode();
+      imgReady = true;
+    } catch (ign) {}
+    if (imgReady && isAwaitingMjpegStream) {
+      setAwaitingMjpegStream(false);
+      this.updateSourceTreeWidth();
+    } else if (!imgReady && !isAwaitingMjpegStream) {
+      setAwaitingMjpegStream(true);
+    }
```

Comment:
what if none of the branches above matches?

Reply:
then the state will not change. `isAwaitingMjpegStream` starts out as true, so a spinner will be shown from the start.

this code handles the situations where:
1. we move from no image data to having image data
2. we move from having image data to no image data

I think that's all the cases we need to handle. I'm basically trying to eliminate spamming the redux state change since this is being called frequently.